services/s3/src/main/java/software/amazon/awssdk/services/s3/extensions/S3ClientSdkExtension.java

```diff
@@ -32,12 +38,36 @@
      * not accessible (e.g., due to access being denied or the bucket existing in another region), an {@link S3Exception} will be
      * thrown.
      *
-     * @param bucketName the bucket to check
+     * @param bucket the bucket to check
      * @return true if the bucket exists and you have permission to access it; false if the bucket does not exist
      * @throws S3Exception if the bucket exists but is not accessible
      */
     @SdkExtensionMethod
-    default boolean doesBucketExist(String bucketName) {
-        return new DefaultS3ClientSdkExtension((S3Client) this).doesBucketExist(bucketName);
+    default boolean doesBucketExist(String bucket) {
+        return new DefaultS3ClientSdkExtension((S3Client) this).doesBucketExist(bucket);
+    }
+
+    /**
+     * Permanently delete a bucket and all of its content, including any versioned objects and delete markers.
+     * <p>
+     * Internally this method will use the {@link S3Client#listObjectsV2(ListObjectsV2Request)} and {@link
+     * S3Client#listObjectVersions(ListObjectVersionsRequest)} APIs to list a bucket's content, buffer keys that are eligible for
+     * deletion into batches of 1000, and delete them in bulk with the {@link S3Client#deleteObjects(DeleteObjectsRequest)} API.
+     * <p>
+     * While this method is optimized to use batch APIs for both listing and deleting, it may not be suitable for buckets
```

Comment:
It may also be useful to direct people to TransferManager for their more complicated needs.

Reply:
Good point, but I don't think we currently plan to support "delete directory" in TM, but it makes sense to support it there.